include/faker-cxx/Number.h

```diff
@@ -1,55 +1,378 @@
+/**
+ * @file Number.h
+ * @authors cieslarmichal, dario-loi
+ * @brief Module that generates random numbers. by wrapping the C++ standard library random number generation utilities.
+ *
+ * @date 2023-07-05
+ *
+ *  @details The module provides functions for both integer and decimal numbers, each with a number of overloads.
+ *  There are generally functions that allow the generation of uniformly distributed numbers within a range,
+ *  and functions that allow the generation of numbers from a given distribution, both in a given range and not.
+ * 
+ *  There is a possibility to only specify the right bound of the range, in which case the left bound is assumed to be 0.
+ */
+
```

Comment:
What do you think about documenting only things that users use? I mean not to document each file, but document classes and classes methods.

Reply:
It might be okay, but documenting the files allows the individual headers to have a doxygen documentation, so if one opens up the html docs he can get an overview of what each module does.

If you feel like it is too much clutter you can go ahead and remove the file docs, but I think they are pretty standard in C++